packages/slider/src/useSlider.ts

```diff
@@ -0,0 +1,317 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { composeEventHandlers, KEYS } from '@zendeskgarden/container-utilities';
+import { IUseSliderProps, IUseSliderReturnValue } from './types';
+
+export const SLIDER_MIN = 0;
+export const SLIDER_MAX = 100;
+export const SLIDER_STEP = 1;
+
+export function useSlider<T extends Element = Element, M extends HTMLElement = HTMLElement>({
+  trackRef,
+  minThumbRef,
+  maxThumbRef,
+  min = SLIDER_MIN,
+  max = SLIDER_MAX,
+  defaultMinValue,
+  defaultMaxValue,
+  minValue,
+  maxValue,
+  onChange = () => undefined,
+  step = SLIDER_STEP,
+  jump = step,
+  disabled,
+  rtl,
+  environment
+}: IUseSliderProps<T, M>): IUseSliderReturnValue {
+  const doc = environment || document;
+  const [trackRect, setTrackRect] = useState<DOMRect>({ width: 0 } as DOMRect);
+
+  const init = (initMinValue = min, initMaxValue = max) => {
+    const retVal = {
+      minValue: initMinValue < min ? min : initMinValue,
+      maxValue: initMaxValue > max ? max : initMaxValue
+    };
+
+    if (retVal.maxValue < min) {
+      retVal.maxValue = min;
+    }
+
+    if (retVal.minValue > retVal.maxValue) {
+      retVal.minValue = retVal.maxValue;
+    }
+
+    return retVal;
+  };
+
+  const [state, setState] = useState(init(defaultMinValue, defaultMaxValue));
+  const isControlled =
+    (minValue !== undefined && minValue !== null) || (maxValue !== undefined && maxValue !== null);
+  const position = isControlled ? init(minValue, maxValue) : state;
+  const setPosition = isControlled ? onChange : setState;
+
+  /*
+   * Effects
+   */
+
+  useEffect(() => {
+    const handleResize = debounce(() => {
+      if (trackRef.current) {
+        setTrackRect(trackRef.current.getBoundingClientRect());
+      }
+    }, 100);
+
+    handleResize();
+
+    const win = doc.defaultView || window;
+
+    win.addEventListener('resize', handleResize);
+
+    return () => {
```

Comment:
We should cancel any inbound debounce calls at this point.
```suggestion
    return () => {
      handleResize.cancel();
```